packages/server/src/guards/org-or-course-roles.guard.ts

```diff
@@ -129,9 +129,7 @@ export class OrgOrCourseRolesGuard implements CanActivate {
       }>();
 
     if (!orgCourseUser) {
-      throw new NotFoundException(
-        'This course does not exist in your organization',
-      );
+      return false;
```

Comment:
```suggestion
      throw new NotFoundException(
        'This course does not exist in your organization',
      );
```

Probably should reverse this change. See below:

> The OrgOrCourseRolesGuard.matchOrgRoles() method was throwing a NotFoundException **when users lacked the required organization role**, preventing the course role check from executing. This caused the guard to fail even when users had the required course professor role.

Are you sure? I went through the logic of the query. This error is only thrown if either A: the given course id is not in this organization, B: the user is not in this organization, or C: either the course or organization does not exist. 

If the user is in the organization without the right role, the check slightly below (github wouldn't let me select it for a comment) will run and return false:
```
 if (!roles.includes(orgCourseUser.ou_role)) {
      return false;
    }
```